forms-flow-web/src/components/Modeler/ProcessCreateEdit.js

```diff
@@ -0,0 +1,641 @@
+import React, { useEffect, useState, useRef, useMemo } from "react";
+import "./Modeler.scss";
+import { useSelector, useDispatch } from "react-redux";
+import { useParams } from "react-router-dom";
+import { push } from "connected-react-router";
+import PropTypes from "prop-types";
+import {
+  updateProcess,
+  publish,
+  unPublish,
+  getProcessDetails,
+  createProcess,
+} from "../../apiManager/services/processServices";
+import Loading from "../../containers/Loading";
+import { MULTITENANCY_ENABLED } from "../../constants/constants";
+import { useTranslation } from "react-i18next";
+import {
+  //createNewProcess,
+  //createNewDecision,
+  extractDataFromDiagram,
+} from "../../components/Modeler/helpers/helper";
+import {
+  CustomButton,
+  HistoryIcon,
+  BackToPrevIcon,
+  ConfirmModal,
+  ErrorModal,
+} from "@formsflow/components";
+import { Card } from "react-bootstrap";
+import ActionModal from "../Modals/ActionModal";
+import ExportDiagram from "../Modals/ExportDiagrams";
+import { toast } from "react-toastify";
+import {
+  createXMLFromModeler,
+  validateProcess,
+  compareXML,
+  compareDmnXML,
+  validateDecisionNames,
+} from "../../helper/processHelper";
+import BpmnEditor from "./Editors/BpmnEditor/BpmEditor.js";
+import DmnEditor from "./Editors/DmnEditor/DmnEditor.js";
+import {
+  setProcessData,
+  setProcessDiagramXML,
+} from "../../actions/processActions";
+
+const EXPORT = "EXPORT";
+const CategoryType = { FORM: "FORM", WORKFLOW: "WORKFLOW" };
+
+const ProcessCreateEdit = ({ type }) => {
+  const { processKey, step } = useParams();
+  const isCreate = step === "create";
+  const isBPMN = type === "BPMN";
+  const isDmn = type === "DMN";
+  const diagramType = isDmn ? "DMN" : "BPMN";
```

Comment:
Can we handle the reused variables in this manner ?

const Process = isBPMN 
  ? {
      name: 'Subflow',
      type: 'BPMN',
      route: 'subflow',
      extension: '.bpmn',
      fileType: 'text/bpmn',
    }
  : {
      name: 'Decision Table',
      type: 'DMN',
      route: 'decision-table',
      extension: '.dmn',
      fileType: 'text/dmn',
    };